taskfile/node_git.go

```diff
@@ -127,11 +124,23 @@ func (node *GitNode) ResolveDir(dir string) (string, error) {
 
 func (node *GitNode) CacheKey() string {
 	checksum := strings.TrimRight(checksum([]byte(node.Location())), "=")
-	prefix := filepath.Base(filepath.Dir(node.path))
-	lastDir := filepath.Base(node.path)
+	lastDir := filepath.Base(filepath.Dir(node.path))
+	prefix := filepath.Base(node.path)
 	// Means it's not "", nor "." nor "/", so it's a valid directory
 	if len(lastDir) > 1 {
-		prefix = fmt.Sprintf("%s-%s", lastDir, prefix)
+		prefix = fmt.Sprintf("%s.%s", lastDir, prefix)
+	}
+	return fmt.Sprintf("git.%s.%s.%s", node.URL.Host, prefix, checksum)
+}
+
+func splitPath(u *url.URL) (string, string) {
```

Comment:
```suggestion
func splitGitURLPath(u *url.URL) (string, string) {
```

That's maybe more descriptive.

Reply:
This could potentially be used elsewhere in the future (in which case I imagine it would be moved), so I opted to go for a name that describes what the function does rather than how it is used. I've updated it to `splitURLOnDoubleSlash` for now.